### ISSUE TITLE
StarRocks discards the NOT NULL constraint of the external table when constructing the full schema, leading to the failure of view creation

### ISSUE THREAD

**Issue by elon-X** [StarRocks/starrocks#48181]
When I try to build an asynchronous partitioned view based on an external table (paimon table / hive table) with a partition field (string type), I encounter an error: `ERROR 1064 (HY000): List partition columns must not be nullable in Materialized view for now`, which causes the asynchronous view construction to fail.

**Root Cause:**

1.StarRocks checks whether partition fields have the NOT NULL constraint when building asynchronous views.
![339466daabc75fbca1853dd9fb14efc4](https://github.com/StarRocks/starrocks/assets/34712973/c0b4af22-28bb-4ed0-b471-a6f711e7ebba)

2.Upon checking the schema of the paimon table, the partition field does have the NOT NULL constraint.
![0bccdd8cef3d4148ea0b12e83c9ec25b](https://github.com/StarRocks/starrocks/assets/34712973/ff14d768-cba9-4110-9656-2989baa25f5a)

3.When StarRocks constructs the full schema, it forces all fields to be nullable. This results in an exception being thrown when constructing the view
![f051bf9209752f1710fc9d1071bd0812](https://github.com/StarRocks/starrocks/assets/34712973/4f7fdefb-ed72-4ca9-9cf5-4fcb48f403f6)

Additionally, this issue can be observed by using `DESC tableA;` or `SHOW CREATE TABLE tableA;`

### Steps to reproduce the behavior (Required)
1. desc tableA; // show create table tableA;

### Expected behavior (Required)
<img width="520" alt="image" src="https://github.com/StarRocks/starrocks/assets/34712973/ea4291f2-9ae8-4582-ae49-395fc500e4d4">

### Real behavior (Required)
<img width="517" alt="image" src="https://github.com/StarRocks/starrocks/assets/34712973/c57fee58-cd4b-4e20-b6c7-75f44dc16903">
<img width="989" alt="image" src="https://github.com/StarRocks/starrocks/assets/34712973/d83b2a07-ba5a-4847-b041-747a9733a15d">

### StarRocks version (Required)
StarRocks-3.3.0

**Comment by LiShuMing** [comment]
Materialized view support list partition with nullable column will be supported in 3.3.1/3.3.2.

**Comment by LiShuMing** [comment]
@elon-X Yep, you are right. But for external tables, `not null` constraint maybe exactly right since its strong schema evolution.

You can try StarRocks v3.3.2 again, this issue should be fixed already.
